worlds/kh1/Rules.py

```diff
@@ -0,0 +1,2142 @@
+                                                                                                                                                    has_item(state, player, "Phil Cup") 
+                                                                                                                                                    and has_item(state, player, "Pegasus Cup") 
+                                                                                                                                                    and has_item(state, player, "Hercules Cup") 
+                                                                                                                                                    and has_x_worlds(state, player, 7, options.keyblades_unlock_chests) 
+                                                                                                                                                    and has_defensive_tools(state, player)
+                                                                                                                                                    and has_item(state, player, "Entry Pass")
+                                                                                                                                                 )
+        multiworld.get_location("Hades Cup Defeat Cloud and Leon Event"                                    , player).access_rule = lambda state: (
+                                                                                                                                                    has_item(state, player, "Phil Cup") 
+                                                                                                                                                    and has_item(state, player, "Pegasus Cup") 
+                                                                                                                                                    and has_item(state, player, "Hercules Cup") 
+                                                                                                                                                    and has_x_worlds(state, player, 7, options.keyblades_unlock_chests) 
+                                                                                                                                                    and has_defensive_tools(state, player)
+                                                                                                                                                    and has_item(state, player, "Entry Pass")
+                                                                                                                                                 )
+        multiworld.get_location("Hades Cup Defeat Yuffie Event"                
```

Comment:
According to pycharm, this file violates pep8 2,550 times. If you add this newline, it'll be 2,549 times instead!
Also you should use state.has instead of state.has_all for one item
```suggestion
    multiworld.completion_condition[player] = lambda state: state.has("Victory", player)
    
```

Reply:
I have no idea why github thinks line 2142 is the multiworld.get_location("Hades Cup Defeat Yuffie Event" line